opentelemetry/src/metrics/instruments/mod.rs

```diff
@@ -23,22 +25,22 @@ pub trait AsyncInstrument<T>: Send + Sync {
 }
 
 /// Configuration for building a sync instrument.
-pub struct InstrumentBuilder<'a, T> {
-    meter: &'a Meter,
+pub struct InstrumentBuilder<T> {
+    instrument_provider: Arc<dyn InstrumentProvider + Send + Sync>,
```

Comment:
Need to check the flow, but would this `dyn` has the cost when the hot path is used as below:
```rust
meter.f64_up_down_counter().add()
```

Reply:
No. This is only in the instrument creation time. No change in the hotpath in this PR.

Reply:
oh wait.. if you are asking about "what if someone keeps creating instruments repeatedily?" . Then, yes, the perf is going to be very bad with or without this change. That is why docs should strongly warn against it!

Reply:
Or we can do something like `&'a I where I: InstrumentProvider + Send + Sync`?